### PR TITLE
feat(bytetracker): modify kalman filter model to get better tracking performance

### PR DESCRIPTION
**Pull request by YoshiRi**
## Description

Previous version has problem in tracking occluded objects.
This is due to the slow aspect ratio estimation.

This PR fixed the problem and some part of the implementation.

|previous|this PR|
|--|--|
|image:w400|image|

- key changes
  - kalman filter implementation(use autoware common package)
  - state choice `top-left-aspectratio-size` to `top-left-width-height`
  - make kalman filter tuning parameters controllable by yaml files

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 7765d48</samp>

This pull request modifies the `bytetrack` package to use the `kalman_filter` package for tracking the state of each track. It adds a yaml file for the Kalman filter parameters and installs it with the package. It also refactors the `STrack` and `byte_tracker` classes to use the `kalman_filter` methods and removes the unused `byte_kalman` code. It updates the package dependencies and the README.md accordingly.

## Related links

[TIER IV internal link](https://tier4.atlassian.net/wiki/spaces/~62d0f5dcafe495359d9e364c/pages/2949548292/Bytetrack)

## Tests performed

Tested with odaiba rosbag.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/bytetrack/config/bytetracker.param.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Kalman filter parmeters for 2d tracking
```
Comment (miursh) [inline perception/bytetrack/config/bytetracker.param.yaml:1]: miss-spelling
```suggestion
# Kalman filter parameters for 2d tracking
```

**Comment by miursh** [comment]
@YoshiRi @manato 
If possible, I would like you to change the video in the [README](https://github.com/autowarefoundation/autoware.universe/blob/8e7e65787ac41f8916a6ba5300a85381d62b1ea2/perception/bytetrack/README.md).

**Review by miursh — APPROVED** [review]
LGTM

**Review by manato — APPROVED** [review]
LGTM. Thank you for your great contribution!

**Pull request merged**